source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -816,10 +822,19 @@ void Filter::onMaxDownstreamConnectionDuration() {
 }
 
 void Filter::onAccessLogFlushInterval() {
-  for (const auto& access_log : config_->accessLogs()) {
-    access_log->log(nullptr, nullptr, nullptr, getStreamInfo());
+  if (!getStreamInfo().streamState()) {
```

Comment:
Hmm, the timer is enabled on the downstream connect event, but the state is set on the upstream connect event. If the delay is longer than the flush interval, it looks like the timer will not be enabled ever again? Is this still a bit buggy?